https://gitlab.com/libtiff/libtiff/-/merge_requests/334

From e7605b93b12c2bf3c864910c23ac976045b5a05a Mon Sep 17 00:00:00 2001
From: Sam James <sam@gentoo.org>
Date: Sat, 21 May 2022 01:01:35 +0100
Subject: [PATCH 1/2] test (autotools): skip script tests if tools aren't built

In Gentoo, we avoid building the tools for multilib (32-bit, x86) builds on
amd64/x86_64 because we only need the library to keep binary applications working.

This causes a test failure in e.g. tiffcp-thumbnail.sh as the 'thumbnail'
binary isn't built. Skip it if unavailable.

Fixes: https://gitlab.com/libtiff/libtiff/-/issues/421
--- a/test/Makefile.am
+++ b/test/Makefile.am
@@ -55,13 +55,14 @@ XFAIL_TESTS =
 CLEANFILES = test_packbits.tif o-*
 
 if HAVE_JPEG
+if TIFF_TOOLS
 JPEG_DEPENDENT_CHECK_PROG=raw_decode
 JPEG_DEPENDENT_TESTSCRIPTS=\
 	tiff2rgba-quad-tile.jpg.sh \
 	tiff2rgba-ojpeg_zackthecat_subsamp22_single_strip.sh \
 	tiff2rgba-ojpeg_chewey_subsamp21_multi_strip.sh \
 	tiff2rgba-ojpeg_single_strip_no_rowsperstrip.sh
-
+endif
 else
 JPEG_DEPENDENT_CHECK_PROG=
 JPEG_DEPENDENT_TESTSCRIPTS=
@@ -76,6 +77,7 @@ check_PROGRAMS = \
 endif
 
 # Test scripts to execute
+if TIFF_TOOLS
 TESTSCRIPTS = \
 	ppm2tiff_pbm.sh \
 	ppm2tiff_pgm.sh \
@@ -156,6 +158,9 @@ TESTSCRIPTS = \
 	testfax4.sh \
 	testdeflatelaststripextradata.sh \
 	$(JPEG_DEPENDENT_TESTSCRIPTS)
+else
+TESTSCRIPTS=
+endif
 
 # This list should contain the references files
 # from the 'refs' subdirectory
-- 
GitLab
